dartsim/src/WorldFeatures.cc

```diff
@@ -80,6 +80,22 @@ const std::string &WorldFeatures::GetWorldCollisionDetector(const Identity &_id)
   return world->getConstraintSolver()->getCollisionDetector()->getType();
 }
 
+/////////////////////////////////////////////////
+void WorldFeatures::SetWorldGravity(
+    const Identity &_id, const LinearVectorType &_gravity)
+{
+  auto world = this->ReferenceInterface<dart::simulation::World>(_id);
+  world->setGravity(_gravity);
+}
+
+/////////////////////////////////////////////////
+WorldFeatures::LinearVectorType WorldFeatures::GetWorldGravity(
```

Comment:
Can we return `const &`?

Reply:
I think we should be careful with returning `const` references because I don't think it works well for values computed in variables on the stack. It works great for accessors to member variables, and indeed dart's `getGravity` method in [dart/simulation/World.hpp](https://github.com/dartsim/dart/blob/v6.10.1/dart/simulation/World.hpp#L124) returns a `const` reference, but changing this API to return a `const` reference would potentially constrain other implementations of this API

I could be wrong in my understanding of c++ here and am open to feedback, but that's my current reasoning for why not to return a `const` reference

Reply:
Ok, that makes sense. I see we have a few const ref returns in `ign-physics` already, but they're mainly returning strings, which I think are safer to assume are coming straight from member variables.